infra/Pos/Communication/Relay/Types.hs

```diff
@@ -44,10 +45,17 @@ instance Buildable PropagationMsg where
     build (DataOnlyPM conts) =
         Buildable.build conts
 
--- | Queue of InvMsges which should be propagated.
-type RelayPropagationQueue = TBQueue (Microsecond, PropagationMsg)
+data RelayContext m = RelayContext
```

Comment:
> Here you are free to change MonadMockable m to whatever constraints you need. You can add MonadIO or MonadReader Something if you want.

This issue is that, since it takes a callback, this `m` must be the same `m` that we use for the entire application, like `RealMode ssc`.

> Is the code with rcfgNextSlotLeader, rcfgDelay, etc. already available somewhere?

`edsko/sketch-relay` in time-warp-nt, but that one is pinned to `IO`. My branch with the generalization hasn't been publish yet.